src/dynamics/se/dp_coupling.F90

```diff
@@ -726,29 +747,27 @@ subroutine derived_phys_dry(cam_runtime_opts, phys_state, phys_tend)
    call cam_thermo_update(const_data_ptr, phys_state%t, pcols, &
         cam_runtime_opts%update_thermodynamic_variables())
 
-   !Call geopotential_t CCPP scheme:
-   call geopotential_t_run(pver, lagrangian_vertical, pver, 1, &
-                           pverp, 1, phys_state%lnpint, phys_state%pint, &
-                           phys_state%pmid, phys_state%pdel, &
-                           phys_state%rpdel, phys_state%t,  const_data_ptr(:,:,ix_q), &
-                           rairv, gravit, zvirv, phys_state%zi, phys_state%zm, pcols, &
-                           errflg, errmsg)
-
-   !NOTE:  Should dry static energy be done in CCPP physics suite? -JN:
+   !Call geopotential_temp CCPP scheme:
+   call geopotential_temp_run(pver, lagrangian_vertical, pver, 1,                        &
+                              pverp, 1, num_advected, phys_state%lnpint,                 &
+                              phys_state%pint, phys_state%pmid, phys_state%pdel,         &
+                              phys_state%rpdel, phys_state%t, const_data_ptr(:,:,ix_q),  &
+                              const_data_ptr, const_prop_ptr, rairv, gravit, zvirv,      &
+                              phys_state%zi, phys_state%zm, columns_on_task, errflg,     &
+                              errmsg)
 
    ! Compute initial dry static energy, include surface geopotential
-   do k = 1, pver
-      do i = 1, pcols
-         phys_state%dse(i,k) = cpair*phys_state%t(i,k) &
-            + gravit*phys_state%zm(i,k) + phys_state%phis(i)
-      end do
-   end do
+   call update_dry_static_energy_run(pver, gravit, phys_state%t, phys_state%zm,          &
+                                     phys_state%phis, phys_state%dse, cpairv,            &
+                                     errflg, errmsg)
 
-!Remove once constituents (and QNEG) are enabled in CAMDEN:
+   ! Ensure tracers are all positive
+   ! Please note this cannot be used until the 'qmin'
+   ! array is publicly provided by the CCPP constituent object. -JN
 #if 0
-      ! Ensure tracers are all positive
-      call qneg3('D_P_COUPLING',lchnk  ,ncol    ,pcols   ,pver    , &
-           1, num_advected, qmin  ,const_data_ptr)
+   call qneg_run('D_P_COUPLING', columns_on_task, pver, &
+                 qmin, const_data_ptr,       &
+                 errflg, errmsg)
 #endif
```

Comment:
It's because this code doesn't work yet in CAM-SIMA (as it needs a new CCPP constituents host-side "get" method), but we will need it to work in order for SIMA to match CAM's scientific results when running with the SE dycore.  The `#if 0` code is designed to both turn this code off for now, and be really noticeable so that we remember to fix it in the future.  If you think there might be a better way to do accomplish that goal then just let me know! 